src/NuGet.Clients/NuGet.CommandLine/Commands/DownloadCommandBase.cs

```diff
@@ -117,7 +122,7 @@ protected IReadOnlyCollection<Configuration.PackageSource> GetPackageSources(Con
             var availableSources = SourceProvider.LoadPackageSources().Where(source => source.IsEnabled);
             var packageSources = new List<Configuration.PackageSource>();
 
-            if (!NoCache)
+            if (!NoCache && !ExcludeCacheAsSource)
```

Comment:
Is "NoCache" also used in other place? why can't just use "NoCache" to exclude cache